### PR TITLE
Fix all_names attribute in SlackConversation, improve test coverage

### PR DESCRIPTION
**Pull request by amckenna-pinterest**

### CODE REVIEW

**Comment by amckenna-pinterest** [comment]
```
>>> c = {
    'id': 'C02KT31H9GX',
    'name': 'TEST-channel',
    'is_channel': True,
    'is_group': False,
    'is_im': False,
    'is_mpim': False,
    'is_private': False,
    'name_normalized': 'test-channel',
    'topic': {'value': '', 'creator': '', 'last_set': 0},
    'previous_names': ['old-test-channel-name', 'some-other-name'],
    'priority': 0
}
>>> from slackminion.slack.conversation import SlackConversation
>>> from unittest import mock
>>> a = mock.Mock()
>>> s = SlackConversation(c, a)
>>> s.name
'TEST-channel'
>>> s.id
'C02KT31H9GX'
>>> s.name_normalized
'test-channel'
>>> s.all_names
['TEST-channel', 'test-channel', 'old-test-channel-name', 'some-other-name']

```

File: slackminion/slack/conversation.py [hunk 1]
```diff
@@ -7,20 +7,24 @@ class SlackConversation(object):
...
 
+    # make arbitrary keys in the dict returned by slack accessible as properties
     def __getattr__(self, item):
         return self.conversation.get(item)
 
     @property
     def all_names(self):
-        return [self.name, self.conversation.get('normalized_name')] + self.conversation.get('previous_names', [])
+        return [self.conversation['name'], self.conversation['name_normalized']] \
+               + self.conversation.get('previous_names', [])
```
Comment (jogo) [inline slackminion/slack/conversation.py:27]: Just to confirm I understand this correctly, the bug here was name_normalized is what is returned by the slack API and not normalized_name?

Comment (amckenna-pinterest) [reply slackminion/slack/conversation.py:27]: No, this was not the bug, but we were using the wrong key for name_normalized.

Comment (jogo) [reply slackminion/slack/conversation.py:27]: Would you mind helping educate me by pointing out the bug to me, as I can't seem to find it on my own. Thank you.

Also thanks for the tests!

Comment (amckenna-pinterest) [reply slackminion/slack/conversation.py:27]: I haven't found the actual bug yet, I'm still working on that.

**Pull request merged**